### PR TITLE
Make OrangeDamException constructor final and add tests

### PR DESCRIPTION
**Pull request by markdorison**
## Description
Make OrangeDamException constructor final and add tests

## Testing Instructions / How This Has Been Tested
🤷🏻 
@ikorgik @adamzimmermann Will this work?

### CODE REVIEW

**Comment by ikorgik** [comment]
I will run the code locally to make sure it works.

**Review by adamzimmermann — APPROVED** [review]
Seems solid to me. Interested in what @ikorgik's testing finds.

**Comment by ikorgik** [comment]
The implementation itself looks good, but it doesn't work as expected in places where we use it:
- https://github.com/ChromaticHQ/orange-dam-php/blob/b75859bba5b48e405693768b201fdc36e17e8e6e/src/Http/Client.php#L86
- https://github.com/ChromaticHQ/orange-dam-php/blob/b75859bba5b48e405693768b201fdc36e17e8e6e/src/Endpoints/OAuth2.php#L42

`OrangeDamException` constructor gets `RequestException` as argument, but we pass text there.

When I implemented it, I thought it's a good idea to have our Exception class to alter messages, I added only basic implementation which needs to be enhanced.

I'd recommend to replace error throw to something like `throw new Exception('Error message');` or rework `OrangeDamException` implementation.

**Comment by markdorison** [comment]
@ikorgik @adamzimmermann I backtracked a bit and made the constructor final instead of replacing the static create method. I think this will work. I also added two tests to catch what @ikorgik  had to test manually.

File: tests/Exception/ClientTest.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+namespace Chromatic\OrangeDam\Http;
+
+use Chromatic\OrangeDam\Exceptions\InvalidArgumentException;
+use Chromatic\OrangeDam\Exceptions\OrangeDamException;
+use GuzzleHttp\Client as GuzzleClient;
+use PHPUnit\Framework\TestCase;
+
+final class ClientTest extends TestCase
+{
+    public function testClientCreation(): void
```
Comment (adamzimmermann) [inline tests/Exception/ClientTest.php]: Can we add some comments that explain what we are testing and why?

**Review by adamzimmermann — COMMENTED** [review]
Very cool. 👏🏼 

Can we document some of this a bit more to help make it clear why we test some of this?

File: tests/Exception/ClientTest.php (hunk 1 above)
Comment (markdorison) [reply tests/Exception/ClientTest.php]: Updated.

**Pull request merged**